File: Andras_Thijs/src/Type.java
import java.util.HashMap;
import java.util.Map;

public enum Type {
    BOOL ("boolean"),
    STRING ("string"),
    INT ("integer"),
    DATA ("date"),
    DECIMAL ("decimal"),
    MONEY ("money");

    private String value;

    Type (String value){
        this.value = value;
    }

    static Map<String, Type> map = new HashMap<String, Type>();

    static {
        for (Type type : Type.values()) {
            map.put(type.value, type);
        }
    }

    public static Type getByCode(String code) {
        return map.get(code);
    }
}

File: Erik_Vincent/Assignment1/Assignment1/Model/QL/AST/Type.cs
using System;

namespace Assignment1.Model.QL.AST
{
    public enum Type
    {
        Boolean,
        Integer,
        String,
        Date,
        Decimal,
        Money,
        Undefined
    }

    public static class TypeMethods
    {
        public static bool IsNumeric(this Type at) => at == Type.Integer || at == Type.Money || at == Type.Decimal;

        public static Type InferArithmeticType(Type leftType, Type rightType)
        {
            if (leftType == Type.Integer)
            {
                switch (rightType)
                {
                    case Type.Integer: return Type.Integer;
                    case Type.Money: return Type.Money;
                    case Type.Decimal: return Type.Decimal;
                }
            }
            if (leftType == Type.Decimal)
            {
                switch (rightType)
                {
                    case Type.Integer: return Type.Decimal;
                    case Type.Decimal: return Type.Decimal;
                    case Type.Money: return Type.Money;
                }
            }
            if (leftType == Type.Money)
            {
                return Type.Money;
            }
            return Type.Undefined;
        }
    }
}
